csharp/test/Drivers/Interop/Snowflake/ErrorHandlingTests.cs

```diff
@@ -0,0 +1,293 @@
+
+            testConfiguration = Utils.LoadTestConfiguration<SnowflakeTestConfiguration>(SnowflakeTestingUtils.SNOWFLAKE_TEST_CONFIG_VARIABLE);
+            testConfiguration.Database = property;
+            yield return new object[] { new SnowflakeTestConfigurationTest(testConfiguration, typeof(AdbcException), ["[Snowflake] 390201 (08004)", "The requested database does not exist or not authorized."]) };
+
+            testConfiguration = Utils.LoadTestConfiguration<SnowflakeTestConfiguration>(SnowflakeTestingUtils.SNOWFLAKE_TEST_CONFIG_VARIABLE);
+            testConfiguration.Warehouse = property;
+            yield return new object[] { new SnowflakeTestConfigurationTest(testConfiguration, typeof(AdbcException), ["[Snowflake] 390201 (08004)", "The requested warehouse does not exist or not authorized."]) };
+        }
+
+        /// <summary>
+        /// A test.
+        /// </summary>
+        public readonly struct SnowflakeTestConfigurationTest
+        {
+            public TestConfiguration TestConfiguration { get; }
+
+            public Type ExceptionType { get; }
+
+            public string[] ExceptionMessageComponents { get; }
+
+            /// <param name="testConfiguration">A test configuration.</param>
+            /// <param name="exceptionType">Expected exception type.</param>
+            /// <param name="exceptionMessageComponents">Expected keywords/phrases to be found in exception message.</param>
+            public SnowflakeTestConfigurationTest(
+                TestConfiguration testConfiguration,
+                Type exceptionType,
+                string[] exceptionMessageComponents)
+            {
+                TestConfiguration = testConfiguration;
+                ExceptionType = exceptionType;
+                ExceptionMessageComponents = exceptionMessageComponents;
+            }
+        }
+
+        private void SelectAndValidateException(
+            string table,
+            string projection,
+            Type expectedExceptionType,
+            string[] expectedExceptionTextContains = null)
+        {
+            Exception actualException = Assert.Throws(expectedExceptionType, () => PerformQuery(table, projection));
+            AssertContainsAll(expectedExceptionTextContains, actualException.Message);
+        }
+
+        private QueryResult PerformQuery(string table, string projection)
+        {
+            string selectStatement = string.Format(
+                "SELECT {0} AS CASTRESULT FROM {1};",
+                projection,
+                table);
+            _output.WriteLine(selectStatement);
+            _statement.SqlQuery = selectStatement;
+            return _statement.ExecuteQuery();
+        }
+
+        private static void AssertContainsAll(string[] expectedTexts, string actualText)
```

Comment:
This seems general enough to put in a common test utilities class.

Reply:
Moved to Utility class.